src/shared/openssl-util.c

```diff
@@ -294,62 +319,59 @@ int ecc_pkey_from_curve_x_y(
 
         _cleanup_(EC_GROUP_freep) EC_GROUP *group = EC_GROUP_new_by_curve_name(curve_id);
         if (!group)
-                return log_debug_errno(SYNTHETIC_ERRNO(EOPNOTSUPP),
-                                       "ECC curve id %d not supported.", curve_id);
+                return log_openssl_errors("ECC curve id %d not supported", curve_id);
 
         _cleanup_(EC_POINT_freep) EC_POINT *point = EC_POINT_new(group);
         if (!point)
                 return log_oom_debug();
 
         if (!EC_POINT_set_affine_coordinates(group, point, bn_x, bn_y, NULL))
-                return log_debug_errno(SYNTHETIC_ERRNO(EIO), "Failed to set ECC coordinates.");
+                return log_openssl_errors("Failed to set ECC coordinates");
 
 #if OPENSSL_VERSION_MAJOR >= 3
         if (EVP_PKEY_fromdata_init(ctx) <= 0)
-                return log_debug_errno(SYNTHETIC_ERRNO(ENOTRECOVERABLE),
```

Comment:
is it safe to change from ENOTRECOVERABLE to EIO? same for EOPNOTSUPP above

Reply:
I think so; the only (current) caller is in tpm2-util and it doesn't check the error value, just passes it along.